src/OpenDebugAD7/AD7DebugSession.cs

```diff
@@ -1277,14 +1282,22 @@ protected override void HandleStackTraceRequestAsync(IRequestResponder<StackTrac
                             textPosition = TextPositionTuple.GetTextPositionOfFrame(m_pathConverter, frame) ?? TextPositionTuple.Nil;
                         }
 
-                        response.StackFrames.Add(new ProtocolMessages.StackFrame()
+                        // remember the state for the lowermost frame
+                        if (startFrame == 0 && i == 0)
+                            m_currentFrameHasSourceCode = textPosition.Source?.Path != null;
```

Comment:
>textPosition.Source [](start = 58, length = 19)

I **think** your goal is to have VS Code make a source request to obtain the disassembly of the frame if there is no source. Is that correct? If so, don't you need to make a source object here that has a sourceReference?